### ISSUE TITLE
Dask spills to disk despite disabling in config

### ISSUE THREAD

**Issue by sridge** [dask/dask#3565]
I set up my Dask config file (```.dask/config.yaml```) [as directed](http://dask.pydata.org/en/latest/setup/hpc.html) to prevent spilling to disk:

```bash
# Fractions of worker memory at which we take action to avoid memory blowup
# Set any of the lower three values to False to turn off the behavior entirely
worker-memory-target: false  # don't spill to disk
worker-memory-spill: false  # don't spill to disk
worker-memory-pause: 0.80  # fraction at which we pause worker threads
worker-memory-terminate: 0.95  # fraction at which we terminate the worker
```

I then tested it:

```python
import dask.array as dsa
da_random = dsa.random.random((1000,100), chunks=(100,100))
da_random.sum().compute()
```

However it continues to spill to disk and returns this error:

```python
FileNotFoundError: [Errno 2] No such file or directory: '/glade2/scratch2/sridge/dask-worker-space/worker-7m1kawhd/storage/%28%27da.random.random_sample-sum-299ed814ecf694436bd7550aba4ced92%27%2C%200%2C%200%29'
```

I am currently using Dask on NCAR's Cheyenne using the latest version of Dask

**Comment by mrocklin** [comment]
Thanks for the bug report.  This is fixed, I think, in https://github.com/dask/distributed/pull/2031

**Comment by apatlpo** [comment]
Are we supposed to modify these lines in `~/.config/dask/distributed.yaml` instead with recent versions of dask and distributed ?

If yes, the dask and distributed docs may need adjustment [here](http://dask.pydata.org/en/latest/setup/hpc.html) and [here](http://distributed.readthedocs.io/en/latest/worker.html#memory-management)

**Comment by mrocklin** [comment]
Ah, nice catch @apatlpo !  Do you have any interest in submitting a PR with a fix?

**Comment by apatlpo** [comment]
not right now, maybe tomorrow.

**Comment by nabinkhadka** [comment]
@mrocklin Was that ever fixed?

**Comment by mrocklin** [comment]
It looks like the bug was fixed, but I'm not sure if the documentation was fixed.  If it was fixed then the fix didn't cross reference this issue.  The best way to find out is probably to go and look directly.

**Comment by GFleishman** [comment]
Is it equivalent to set these spill related variables like so:

```python
from dask.distributed import Client, LocalCluster
from dask-jobqueue import LSFCluster
import dask.config

dask.config.set(
    {'distributed.worker.memory.target':False,
     'distributed.worker.memory.spill':False,
     'distributed.worker.memory.pause':0.8,
     'distributed.worker.memory.terminate':0.95,}
)

# set up LocalCluster or dask-jobqueue cluster
```

or does it need to be in a config file? The [documentation here](https://jobqueue.dask.org/en/latest/configuration-setup.html#no-local-storage) is a little bit confusing for me.

**Comment by GFleishman** [comment]
Additionally - after setting configs as above, I still see:

```distributed.worker - WARNING - Worker is at 81% memory usage. Pausing worker.  Process memory: 24.32 GB -- Worker memory limit: 30.00 GB```

In the `.err` output for my workers.

**Comment by jsignell** [comment]
It's equivalent to set the config in the as you have written, but you have invalid values for `distributed.worker.memory.pause` and `distributed.worker.memory.terminate` those should be floats as indicated in the example in the docs.

**Comment by MatthewLennie** [comment]
@jsignell @GFleishman did that end up working for you? If so, any hints?

**Comment by MatthewLennie** [comment]
One thing I just tested with the pattern in @GFleishman's example above was whether the config settings are actually reaching the worker. 


```
dask.config.set({'distributed.worker.memory.target': False, 
    'distributed.worker.memory.spill': False, 
    'distributed.worker.memory.pause':0.95,
    'distributed.worker.memory.terminate':0.96}
    )


cluster = SLURMCluster(
    project="#######",
    cores=28,  # Total number of cores per job
    memory="110GiB",  # Total amount of memory per job
    processes=1,  # Number of Python processes per job
    interface="ib0",  # Network interface to use like eth0 or ib0
    death_timeout=360,  # Seconds to wait if a worker loses a scheduler
    shebang="#!/usr/bin/env bash",
    queue="######",
    walltime="00:29:00",
    extra=[
        "--memory-limit auto",
    ],
    job_extra=["--gpus-per-node=1"],  # This ensures that the batch job includes a request for a GPU.
)
print(dask.config.config)

# Create x jobs
cluster.scale(1 * 1)
client = Client(cluster)
def get_config():
     return dask.config.config
     
client.run(get_config)
```

which returned the default settings: 

```
'memory':{'recent_to_old_time': '30s', 'target': 0.6, 'spill': 0.7, 'pause': 0.8, 'terminate': 0.95}
```

It appears to me that the configs aren't reaching the workers.  I tried instead 

```
def disable_spill():
    dask.config.set({'distributed.worker.memory.target': False, 
    'distributed.worker.memory.spill': False, 
    'distributed.worker.memory.pause':0.95,
    'distributed.worker.memory.terminate':0.96}
    )
    print(dask.config.config)
    
client.register_worker_callbacks(setup=disable_spill)

```
which sets the worker config as expected. 
```
'memory':{'recent_to_old_time': '30s', 'target': False, 'spill': False, 'pause': 0.95, 'terminate': 0.96}
```
@mrocklin It seems to me that the config from that I have set hasn't been inherited and communicated through to the cluster. Bug or feature?

**Comment by GFleishman** [comment]
I reached the same conclusion as @MatthewLennie - config settings set dynamically through `dask.config` in the client process are not inherited by workers set up later by `dask-jobqueue` clusters (in my case it's`LSFCluster`). I've changed my design to just write a yaml to ~/.config/dask/ before creating the cluster. The essentials are:

```python
import dask.config
from dask_jobqueue import LSFCluster

options = { /* some dict of config settings */ }
dask.config.set(options)
yaml_path = "/PATH/TO/.config/dask/my_config_settings.yaml"
with open(yaml_path, 'w') as f:
    yaml.dump(dask.config.config, f, default_flow_style=False)  # can't remember why I needed flow style flag

cluster = LSFCluster(
...
)
```

In my actual code I've wrapped the `LSFCluster` object and defined my own dunder methods - so these config changes, and the longterm persistence of the yaml file, can be handled at construction/destruction - but that's all up to you of course.
